plugin-build/src/main/kotlin/io/sentry/android/gradle/sourcecontext/UploadSourceBundleTask.kt

```diff
@@ -33,6 +33,9 @@ abstract class UploadSourceBundleTask : Exec() {
             includeSourceContext.getOrElse(false) &&
                 !sourceBundleDir.asFileTree.isEmpty
         }
+
+        // Allows gradle to consider this task up-to-date if the inputs haven't changed
```

Comment:
👋 just a fly on the wall. I'm a sentry user and a stakeholder in issue #616,  but I'm reasonably certain this is incorrect. See the JavaDoc for the `upToDateWhen` method: https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskOutputs.html#upToDateWhen-groovy.lang.Closure-

>  Adds a predicate to determine whether previous outputs of this task can be reused. The given closure is executed at task execution time. The closure is passed the task as a parameter. If the closure returns false, previous outputs of this task cannot be reused and the task will be executed. That means the task is out-of-date and no outputs will be loaded from the build cache.

> You can add multiple such predicates. The task outputs cannot be reused when any predicate returns false. 


See this discussion: https://discuss.gradle.org/t/outputs-uptodatewhen-confusion/17973/3


> If you read “false” as “not”, then…
`outputs.upToDateWhen { true }` means “outputs are up to date”.
`outputs.upToDateWhen { false }` means “outputs are not up to date”.

> If any upToDateWhen spec returns false, the task is considered out of date. If they return true, Gradle does the normal behavior of checking input/output files.

This should be removed to avoid confusion, but I don't think it will negatively impact anything